cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java

```diff
@@ -84,13 +100,17 @@ public void enforce(EntityId entity, Principal principal, Set<Action> actions) t
 
   @Override
   public Predicate<EntityId> createFilter(Principal principal) throws Exception {
-    if (!authorizationEnabled) {
+    if (!isSecurityAuthorizationEnabled()) {
       return ALLOW_ALL;
     }
     // For accessing system datasets for internal operations like recording metadata, usage, lineage, etc
     if (Principal.SYSTEM.equals(principal)) {
       return ALLOW_ALL;
     }
+    // If the principal is a super user, do not filter
+    if (isSuperUser(principal)) {
```

Comment:
We'll be able to do so once https://issues.cask.co/browse/CDAP-6873 is resolved.